client/src/screens/events/Detail.js

```diff
@@ -56,34 +59,77 @@ class Detail extends Component {
     });
     return mapMarkers;
   }
-  static makeResponseMarkers(responses) {
+  static makeResponseMarkers(userid, responses) {
     const mapMarkers = [];
-
-    responses.forEach((r) => {
-      if (r.locationLatitude !== null && r.locationLongitude !== null) {
-        mapMarkers.push({
-          displayName: r.user.displayName,
-          status: r.status,
-          destination: r.destination && r.destination.name,
-          id: r.user.username,
-          latitude: r.locationLatitude,
-          longitude: r.locationLongitude,
-        });
-      }
-    });
+    if (responses) {
+      responses.forEach((r) => {
+        if (r.locationLatitude !== null && r.locationLongitude !== null) {
+          if (r.user.id !== userid) {
+            mapMarkers.push({
+              displayName: r.user.displayName,
+              status: r.status,
+              destination: r.destination && r.destination.name,
+              locationTime: moment.unix(r.locationTime).fromNow(),
+              id: r.user.username,
+              latitude: r.locationLatitude,
+              longitude: r.locationLongitude,
+            });
+          }
+        }
+      });
+    }
     return mapMarkers;
   }
 
+  constructor(props) {
+    super(props);
+    this.mounted = false;
+    this.state = {
```

Comment:
if mounted is not used remove the constructor and create state outside constructor:
```
  state = {
    ...
  }
```
if mounted is used maybe just init that outside constructor as well